external-crates/move/move-compiler/src/diagnostics/codes.rs

```diff
@@ -14,11 +14,16 @@ pub enum Severity {
     Bug = 3,
 }
 
+#[derive(PartialEq, Eq, Clone, Copy, Debug, Hash, PartialOrd, Ord)]
+pub struct DiagnosticsID {
+    pub category: u8,
+    pub code: u8,
+}
+
 #[derive(PartialEq, Eq, Clone, Debug, Hash)]
 pub struct DiagnosticInfo {
     severity: Severity,
-    category: u8,
-    code: u8,
+    id: DiagnosticsID,
     message: &'static str,
     external_prefix: Option<&'static str>,
```

Comment:
We could maybe fold in `external_prefix: Option<&'static str>,` into `DiagnosticsID` as a solution to the domain separation problem

Reply:
It's a good start but we need to domain-separate the `All` filter and the `Category` filter as well. I followed a similar path to add `external_prefix` to both and added extra tests for checking this.